### PR TITLE
Delete Event User Story

### PR DESCRIPTION
**Pull request by RoyceLuo1216**
Completed work on edit/delete view, and finished the delete event use case. 

note, there are currently imports on entities in editDeleteView, however, these are only used in the main function purely used to visualize and test the view for the interim. These imports will be deleted at finalization, to abide by CA.

further note, since we dont have a main class we can't use cardlayout for managing transitions between views yet, so that logic needs to be implemented still, but most of the logic is there already.

### CODE REVIEW

File: src/main/java/entities/EventEntity/FlexibleEvent.java
Comment (venniseho) [inline src/main/java/entities/EventEntity/FlexibleEvent.java:1]: I think we can delete this file, no?

**Review by venniseho — CHANGES_REQUESTED** [review]
- let's not merge the incomplete files into main yet (i.e. schedule controller/presenter)
- I know we've kind of integrated your use case (delete) and mine (edit) for ease of use, but I think we should split the files for edit and delete to make it more in line with CA and SOLID principles (specifically, the single responsibility principle), but we can talk about it more
- otherwise, everything looks relatively good in relation to design principles
- please add comments! (i.e. javadoc for your methods)

File: src/main/java/entities/EventEntity/FlexibleEvent.java
Comment (RoyceLuo1216) [reply src/main/java/entities/EventEntity/FlexibleEvent.java:1]: Yea deleted

**Comment by utcjcao** [comment]
1. To fix external clean architecture issues:

a) rename data_access Schedule into InMemoryDataAccessObject
b) delete data package
c) write data access interface for: delete, edit, chatbot conflicts
d) merge chris dao branch with data access branch to get the json dao (inmemory, json dao are written)

2. To fix chatbot event conflict view model:
a) update message to reflect separation of time estimation and event conflict chatbots
b) update ui to include access to time estimation 
c) merge with chris-branch-dao-v2 for time estimation usecase

3. delete event conflict output data, its unused

File: src/main/java/adapter/CohereParser.java
Comment (utcjcao) [inline src/main/java/adapter/CohereParser.java:1]: delete this file, its unused

File: src/main/java/data/EventStorage.java
Comment (utcjcao) [inline src/main/java/data/EventStorage.java:1]: this file is renamed in main

Comment (utcjcao) [reply src/main/java/data/EventStorage.java:1]: also don't forget to implement the data, edit, chatbot conflict interfaces when we make our daos

File: src/main/java/usecase/chatbot_event_conflict/EventConflictInteractor.java
Comment (utcjcao) [inline src/main/java/usecase/chatbot_event_conflict/EventConflictInteractor.java:1]: can the cohere api be referenced here? i need to double check

File: src/main/java/usecase/delete/DeleteEventInteractor.java
Comment (utcjcao) [inline src/main/java/usecase/delete/DeleteEventInteractor.java:1]: if needed, add exception cases if an error occurs

File: src/main/java/usecase/edit/EditInputData.java
Comment (utcjcao) [inline src/main/java/usecase/edit/EditInputData.java:1]: add some spaces separating lines (for checkstyle)

File: src/test/java/usecase/deleteevent/DeleteEventIntegrationTest.java
Comment (utcjcao) [inline src/test/java/usecase/deleteevent/DeleteEventIntegrationTest.java:1]: might need to format for newlines

File: src/test/java/usecase/deleteevent/DeleteEventPresenterTest.java
Comment (utcjcao) [inline src/test/java/usecase/deleteevent/DeleteEventPresenterTest.java:1]: watch out for newlines formatting. mightneed to trim

File: src/test/java/usecase/edit/EditInteractorTest.java
Comment (utcjcao) [inline src/test/java/usecase/edit/EditInteractorTest.java:1]: might need to check for empty lists

**Review by utcjcao — COMMENTED** [review]
1. To fix external clean architecture issues:
a) rename data_access Schedule into InMemoryDataAccessObject
b) delete data package
c) write data access interface for: delete, edit, chatbot conflicts
d) merge chris dao branch with data access branch to get the json dao (inmemory, json dao are written)

2. To fix chatbot event conflict view model:
a) update message to reflect separation of time estimation and event conflict chatbots
b) update ui to include access to time estimation
c) merge with chris-branch-dao-v2 for time estimation usecase

3. delete event conflict output data, its unused


some tests might need to watch out for newlines, trimming, edges cases for empty lists

also need to remove cohereparser (redundant)

**Review by venniseho — APPROVED** [review]
All the changes look good!!

**Review by utcjcao — CHANGES_REQUESTED** [review]
code looks good to merge 

delete cohere parser

**Review by utcjcao — APPROVED** [review]
amazing <3

**Pull request merged**